modules/apps/29-fee/ica_test.go

```diff
@@ -0,0 +1,192 @@
+
+	packetID := channeltypes.NewPacketId(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 1)
+	packetFees, found := suite.chainA.GetSimApp().IBCFeeKeeper.GetFeesInEscrow(suite.chainA.GetContext(), packetID)
+	suite.Require().True(found)
+	suite.Require().Equal(expectedFee, packetFees.PacketFees[0].Fee)
+
+	interchainAccountAddr, found := suite.chainB.GetSimApp().ICAHostKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), ibctesting.FirstConnectionID, path.EndpointA.ChannelConfig.PortID)
+	suite.Require().True(found)
+
+	// fund the interchain account on chainB
+	coins := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100000)))
+	msgBankSend := &banktypes.MsgSend{
+		FromAddress: suite.chainB.SenderAccount.GetAddress().String(),
+		ToAddress:   interchainAccountAddr,
+		Amount:      coins,
+	}
+
+	res, err = suite.chainB.SendMsgs(msgBankSend)
+	suite.Require().NotEmpty(res)
+	suite.Require().NoError(err)
+
+	// prepare a simple stakingtypes.MsgDelegate to be used as the interchain account msg executed on chainB
+	validatorAddr := (sdk.ValAddress)(suite.chainB.Vals.Validators[0].Address)
+	msgDelegate := &stakingtypes.MsgDelegate{
+		DelegatorAddress: interchainAccountAddr,
+		ValidatorAddress: validatorAddr.String(),
+		Amount:           sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(5000)),
+	}
+
+	data, err := icatypes.SerializeCosmosTx(suite.chainA.GetSimApp().AppCodec(), []sdk.Msg{msgDelegate})
+	suite.Require().NoError(err)
+
+	icaPacketData := icatypes.InterchainAccountPacketData{
+		Type: icatypes.EXECUTE_TX,
+		Data: data,
+	}
+
+	// ensure chainB is allowed to execute stakingtypes.MsgDelegate
+	params := icahosttypes.NewParams(true, []string{sdk.MsgTypeURL(msgDelegate)})
+	suite.chainB.GetSimApp().ICAHostKeeper.SetParams(suite.chainB.GetContext(), params)
+
+	// build the interchain accounts packet
+	packet := buildInterchainAccountsPacket(path, icaPacketData.GetBytes(), 1)
+
+	// write packet commitment to state on chainA and commit state
+	commitment := channeltypes.CommitPacket(suite.chainA.GetSimApp().AppCodec(), packet)
+	suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.SetPacketCommitment(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 1, commitment)
+	suite.chainA.NextBlock()
+
+	err = path.RelayPacket(packet)
+	suite.Require().NoError(err)
+
+	// ensure escrowed fees are cleaned up
+	packetFees, found = suite.chainA.GetSimApp().IBCFeeKeeper.GetFeesInEscrow(suite.chainA.GetContext(), packetID)
+	suite.Require().False(found)
+	suite.Require().Empty(packetFees)
+
+	// assert the value of the account balance after fee distribution
+	postDistBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+	suite.Require().Equal(preEscrowBalance, postDistBalance)
```

Comment:
Hmmm, I am not sure I understand how it's possible that `postDistBalance` matches the `preEscrowBalance`... The former should have the recv and ack fees deducted but the timeout fee refunded and the latter should have all three (ack, recv, timeout) fees deducted, right?

Reply:
Ah yes, its because `suite.chainA.SenderAccount.GetAddress()` is used above as the `RefundAddress` arg when the `msgPayPacketFee` is created. 
And following this, when `path.RelayPacket()` is called on L166, it is the same sender account used as the `signer` for acknowledging the packet. [See here](https://github.com/cosmos/ibc-go/blob/main/testing/endpoint.go#L445), this function gets invoked as part as of the `RelayerPacket()` flow.

Reply:
I have adjusted the test based on your comment here as I thought all fees going to and from the same account was obfuscating things a little.

I've added `SetCounterpartyAddress` on chainB above, and registered `chainA.SenderAccounts[1]` as the address for recv fee distribution, and also added an inline comment here to explain why timeout fees and ack fees go to the same address (chainA.SenderAccount). 
Essentially we are limited by the APIs of the testing library and the sender account must be used as the signer/refund address of `msgPayPacketFee` as that's being delivered via `chain.SendMsgs` which [signs using the default chain sender account](https://github.com/cosmos/ibc-go/blob/main/testing/chain.go#L311-L321).

Along with that when `path.RelayPacket(packet)` is called it uses the default sender account of chainA and chainB for acks and recvs respectively. To the ack fee will be paid out to `chainA.SenderAccount`, i.e. the "relayer" of the msg in this case.

I think this works okay, I could maybe try to work around the refund address issue for timeout fee by calling into the keeper funcs directly and bypassing `chain.SendMsgs()`, let me know what you think.

Reply:
This is already good. Thanks, @damiannolan!